File: xamarin/PaymentDeeplinkDemo/MainActivity.cs
using Android.App;
using Android.OS;
using Android.Support.V7.App;
using Android.Widget;
using System.Web;
using Android.Content;
using System;

namespace PaymentDeeplinkDemo
{
    [Activity(Label = "@string/app_name", Theme = "@style/AppTheme", MainLauncher = true)]
    [IntentFilter(new[] { Android.Content.Intent.ActionView },
              Categories = new[] { Android.Content.Intent.CategoryBrowsable, Android.Content.Intent.CategoryDefault },
              DataScheme = "xamarindeeplink",
              DataHost = "pay-response",
              AutoVerify = true
        )]
    public class MainActivity : AppCompatActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            Xamarin.Essentials.Platform.Init(this, savedInstanceState);
            SetContentView(Resource.Layout.activity_main);

            Button button = FindViewById<Button>(Resource.Id.deepLinkButton);
            TextView resultTextView = FindViewById<TextView>(Resource.Id.deepLinkTextView);

            Android.Net.Uri.Builder uriBuilder = new Android.Net.Uri.Builder();
            uriBuilder.Authority("pay");
            uriBuilder.Scheme("payment-app");

            uriBuilder.AppendQueryParameter("amount", "200");
            uriBuilder.AppendQueryParameter("editable_amount", "0"); // 1 = true and 0 = false
            uriBuilder.AppendQueryParameter("transaction_type", "credit"); //DEBIT, CREDIT, VOUCHER
            uriBuilder.AppendQueryParameter("installment_type", "merchant"); //MERCHANT, ISSUER, NONE
            uriBuilder.AppendQueryParameter("return_scheme", "xamarindeeplink");
            
            String order_id = null;
            if (order_id != null)
            {
                uriBuilder.AppendQueryParameter("order_id", order_id);
            }

            String installment_count = "2";
            if (installment_count != null)
            {
                uriBuilder.AppendQueryParameter("installment_count", installment_count); // 2 to 99
            }

            try
            {
                if (Intent.Data != null)
                {
                    resultTextView.Text = Intent.DataString;
                }
            }catch(Exception e)
            {
                resultTextView.Text = e.Message;
            }
            
            button.Click += delegate
            {
                Intent intent = new Intent(Intent.ActionView);
                intent.AddFlags(ActivityFlags.NewTask);
                intent.SetData(uriBuilder.Build());
                StartActivity(intent);
            };
        }
    }
}

File: native/app/src/main/java/br/com/stone/posandroid/posdeeplink/MainActivity.kt
package br.com.stone.posandroid.posdeeplink

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import java.lang.Exception

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        deepLink.setOnClickListener {
            startPayment()
        }
    }

    private fun startPayment() {
        val uriBuilder = Uri.Builder()
        uriBuilder.authority("pay")
        uriBuilder.scheme("payment-app")
        uriBuilder.appendQueryParameter(RETURN_SCHEME, "deeplinktest")
        uriBuilder.appendQueryParameter(THIRD_PARTY_THEME_ENABLED, "true")
        uriBuilder.appendQueryParameter(THIRD_PARTY_LOGO, "ic_activation")

        if (editTextAmount.text.toString().isNotBlank()) {
            val amount = editTextAmount.text.toString()
            uriBuilder.appendQueryParameter(AMOUNT, amount)
        }

        val editableAmount = if (editableAmountCheckbox.isChecked) "1" else "0"
        uriBuilder.appendQueryParameter(EDITABLE_AMOUNT, editableAmount)

        val transactionType: String? = when (spinnerTransactionType.selectedItemPosition) {
            0 -> "DEBIT"
            1 -> "CREDIT"
            2 -> "VOUCHER"
            3 -> "INSTANT_PAYMENT"
            4 -> "PIX"
            else -> null
        }
        if (transactionType != null) {
            uriBuilder.appendQueryParameter(TRANSACTION_TYPE, transactionType)
        }

        val installmentType: String? = when (spinnerInstallmentType.selectedItemPosition) {
            0 -> "MERCHANT"
            1 -> "ISSUER"
            2 -> "NONE"
            else -> null
        }
        if (installmentType != null) {
            uriBuilder.appendQueryParameter(INSTALLMENT_TYPE, installmentType)
        }

        if (editTextInstallmentCount.text.toString().isNotBlank()) {
            val installmentCount = editTextInstallmentCount.text.toString()
            uriBuilder.appendQueryParameter(INSTALLMENT_COUNT, installmentCount)
        }

        if (editTextOrderId.text.toString().isNotBlank()) {
            val orderId = editTextOrderId.text.toString()
            uriBuilder.appendQueryParameter(ORDER_ID, orderId)
        }

        val intent = Intent(Intent.ACTION_VIEW)
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        intent.data = uriBuilder.build()
        startActivity(intent)

        Log.v(TAG, "toUri(scheme = ${intent.data})")
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        try {
            Log.i("onNewIntent", intent?.data.toString())
            if (intent?.data != null) {
                Toast.makeText(this, intent.data.toString(), Toast.LENGTH_LONG).show()
                Log.i("DeeplinkPay Response", intent.data.toString())
            }
        } catch (e: Exception) {
            Toast.makeText(this, e.toString(), Toast.LENGTH_LONG).show()
            Log.e("Deeplink error", e.toString())
        }
    }

    fun btnDeepLinkCancel(view: View) {
        val uriBuilder = Uri.Builder()
        uriBuilder.authority("cancel")
        uriBuilder.scheme("cancel-app")
        uriBuilder.appendQueryParameter(RETURN_SCHEME, "deeplinktest")
        uriBuilder.appendQueryParameter(THIRD_PARTY_THEME_ENABLED, "true")
        uriBuilder.appendQueryParameter(THIRD_PARTY_LOGO, "ic_activation")

        //val intent = Intent(this, CancelActivity::class.java)
        val intent = Intent(Intent.ACTION_VIEW)
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        intent.data = uriBuilder.build()

        startActivity(intent)
    }

    fun btnDeeplinkPrinter(view: View) {
        val intent = Intent(this, PrinterActivity::class.java)
        startActivity(intent)
    }

    fun btnDeeplinkReprinter(view: View) {
        val intent = Intent(this, ReprinterActivity::class.java)
        startActivity(intent)
    }

    companion object {
        private const val AMOUNT = "amount"
        private const val ORDER_ID = "order_id"
        private const val EDITABLE_AMOUNT = "editable_amount"
        private const val TRANSACTION_TYPE = "transaction_type"
        private const val INSTALLMENT_TYPE = "installment_type"
        private const val INSTALLMENT_COUNT = "installment_count"
        private const val RETURN_SCHEME = "return_scheme"
        private const val THIRD_PARTY_THEME_ENABLED = "third_party_theme_enabled"
        private const val THIRD_PARTY_LOGO = "third_party_logo"
        const val TAG = "SendDeeplinkPayment"
    }
}
